test/TestHelpers.jl

```diff
@@ -102,9 +100,15 @@ function challenge_prompt(cmd::Cmd, challenges; timeout::Integer=10, debug::Bool
         finally
             kill(p)
         end
-        # Determine if the process was explicitly killed
-        killed = process_exited(p) && (p.exitcode != 0 || p.termsignal != 0)
-        killed && error("Too many prompts. $(format_output(out))")
+
+        # Process timed out or aborted
+        if !success(p)
+            if istaskdone(t)
```

Comment:
Looking at this again there is a small timing window where the process could be killed externally and the time out task could end